File: e-elearning-main/h-learning-fe-admin/src/dto/request/quizz.ts
import { ENTITY_TYPE, RESULT_TYPE } from "@/model/quizz"

export type CreateQuizzRequest = {
  ask: string
  resultType: RESULT_TYPE
  result: string[]
  option: string[]
  time: number
  entityType: ENTITY_TYPE
  entityId: number
}

export type UpdateQuizzRequest = {
  id: number
  ask: string
  resultType: RESULT_TYPE
  result: string[]
  option: string[]
  time: number
}

export type DeleteQuizzRequest = {
  id: number
}

File: e-elearning-main/service/internal/dto/client/quizz.go
package requestdata

import (
	"app/internal/entity"
)

type CreateQuizzRequest struct {
	Ask        string             `json:"ask"`
	ResultType entity.RESULT_TYPE `json:"resultType"`
	Result     []string           `json:"result"`
	Option     []string           `json:"option"`
	Time       int                `json:"time"`
	EntityType entity.ENTITY_TYPE `json:"entityType"`
	EntityId   uint               `json:"entityId"`
}

type UpdateQuizzRequest struct {
	Id         uint64             `json:"id"`
	Ask        string             `json:"ask"`
	ResultType entity.RESULT_TYPE `json:"resultType"`
	Result     []string           `json:"result"`
	Option     []string           `json:"option"`
	Time       int                `json:"time"`
}

type DeleteQuizzRequest struct {
	Id uint64 `json:"id"`
}
